Artifacts/windows-azuresdknet-vs2013/InstallViaWebPICmd.ps1

```diff
@@ -0,0 +1,309 @@
+    }
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Writes specified string to the console as well as to the script log (indicated by $ScriptLog).
+#
+# Parameters:
+#  - $message: The string to write.
+#
+# Return:
+#  - N/A.
+#
+# Notes:
+#  - N/A.
+#
+
+function WriteLog
+{
+    Param(
+        <# Can be null or empty #> $message
+    )
+
+    $timestampedMessage = $("[" + [System.DateTime]::Now + "] " + $message) | % {  
+        Out-File -InputObject $_ -FilePath $ScriptLog -Append
+    }
+}
+
+##################################################################################################
+
+# 
+# Description:
+#  - Installs the chocolatey package manager.
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - If installation is successful, then nothing is returned.
+#  - Else a detailed terminating error is thrown.
+#
+# Notes:
+#  - @TODO: Write to $chocolateyInstallLog log file.
+#  - @TODO: Currently no errors are being written to the log file ($chocolateyInstallLog). This needs to be fixed.
+#
+
+function InstallChocolatey
+{
+    Param(
+        [ValidateNotNullOrEmpty()] $chocolateyInstallLog
+    )
+
+    WriteLog "Installing Chocolatey..."
+
+    Invoke-Expression ((new-object net.webclient).DownloadString('https://chocolatey.org/install.ps1')) | Out-Null
+
+    WriteLog "Success."
+}
+
+##################################################################################################
+
+#
+# Description:
+#  - Installs WebPICmd via chocolatey package manager.
+#
+# Parameters:
+#  - N/A.
+#
+# Return:
+#  - If installation is successful, then nothing is returned.
+#  - Else a detailed terminating error is thrown.
+#
+# Notes:
+#  - N/A.
+#
+
+function InstallWebPICmd
+{
+    $package = "webpicmd"
+
+    WriteLog $("Installing chocolatey package: " + $package)
+
+    # install git via chocolatey
+    choco install $package --force --yes --acceptlicense --verbose | Out-Null 
+
+    if ($? -eq $false)
+    {
+        $errMsg = $("Error! Installation failed. Please see the chocolatey logs in %ALLUSERSPROFILE%\chocolatey\logs folder for details.")
+        WriteLog $errMsg
+        Write-Error $errMsg 
+    }
+    
+    WriteLog "Success."
+}
+
+##################################################################################################
+
+#
+# Description:
+#  - Installs the specified prodcut via WebPICmd.
+#
+# Parameters:
+#  - $productId: A semi-colon or comma separated list of products to be installed via WebPI.
+#  - $language: Language of installers that'll be used.
+#
+# Return:
+#  - If installation is successful, then nothing is returned.
+#  - Else a detailed terminating error is thrown.
+#
+# Notes:
+#  - @TODO: If a non-existent package is specified, WebPICmd still returns exit code 0 even though
+#    nothing is installed. 
+#
+
+function InstallProducts

```

Comment:
Should this be plural when it accepts a singular product ID?  This may be clearer with consistency between the cardinality for the name and primary input param.